foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiStreamingHandler.java

```diff
@@ -0,0 +1,98 @@
+package com.sap.ai.sdk.foundationmodels.openai;
+
+import static com.sap.ai.sdk.foundationmodels.openai.OpenAiClient.JACKSON;
+import static com.sap.ai.sdk.foundationmodels.openai.OpenAiResponseHandler.buildExceptionAndThrow;
+import static com.sap.ai.sdk.foundationmodels.openai.OpenAiResponseHandler.parseErrorAndThrow;
+
+import com.sap.ai.sdk.foundationmodels.openai.model.StreamedDelta;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.HttpEntity;
+
+@Slf4j
+@RequiredArgsConstructor
+class OpenAiStreamingHandler<D extends StreamedDelta> {
+
+  @Nonnull private final Class<D> deltaType;
+
+  /**
+   * Processes a {@link ClassicHttpResponse} and returns some value corresponding to that response.
+   *
+   * @param response The response to process
+   * @return A {@link Stream} of a model class instantiated from the response
+   * @throws OpenAiClientException in case of a problem or the connection was aborted
+   */
+  @Nonnull
+  public Stream<D> handleResponse(@Nonnull final ClassicHttpResponse response)
+      throws OpenAiClientException {
+    if (response.getCode() >= 300) {
+      buildExceptionAndThrow(response);
+    }
+    return parseResponse(response);
+  }
+
+  /**
+   * @param response The response to process
+   * @return A {@link Stream} of a model class instantiated from the response
+   * @author stippi
+   */
```

Comment:
```suggestion
```

Reply:
I am legally required to mention Stephan as the author 🫡

Reply:
Why? None of the other authors of this codebase are mentioned.. Not that I have anything against it, it just looks a bit odd IMO 😉

Reply:
Stephan Assmus:
> If possible, please attribute the code to me somehow. :smile: